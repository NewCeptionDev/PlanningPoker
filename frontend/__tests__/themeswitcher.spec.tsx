import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import ThemeSwitcher from 'src/app/themeswitcher'

describe('ThemeSwitcher', () => {
  it('should render correctly', () => {
    const { container } = render(<ThemeSwitcher />)
    expect(container).toMatchSnapshot()
  })

  it('should render all components', () => {
    render(<ThemeSwitcher />)
    expect(screen.getByTestId('theme-switcher')).toBeInTheDocument()
    expect(screen.getByTestId('theme-switcher-toggle')).toBeInTheDocument()
    expect(screen.getByTestId('theme-switcher-label')).toBeInTheDocument()
    expect(screen.getByTestId('theme-switcher-label-background')).toBeInTheDocument()
  })
})
