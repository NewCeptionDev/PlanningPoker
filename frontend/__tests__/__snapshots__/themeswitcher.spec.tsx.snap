// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeSwitcher should render correctly 1`] = `
<div>
  <div
    class="grid place-items-center h-full relative"
    data-testid="theme-switcher"
  >
    <input
      class="hidden toggle--checkbox"
      data-testid="theme-switcher-toggle"
      id="toggle"
      type="checkbox"
    />
    <label
      class="toggle--label"
      data-testid="theme-switcher-label"
      for="toggle"
    >
      <span
        class="toggle--label-background"
        data-testid="theme-switcher-label-background"
      />
    </label>
  </div>
</div>
`;
